addons/l10n_it_edi/wizard/account_move_send.py

```diff
@@ -1,12 +1,216 @@
 # -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from base64 import b64encode
+from markupsafe import Markup
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.addons.l10n_it_edi.tools.xml_utils import format_errors
+from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import AccountEdiProxyError
 
 
 class AccountMoveSend(models.Model):
     _inherit = 'account.move.send'
 
-    def _get_mail_attachment_from_doc(self, doc):
-        if doc.edi_format_id.code == 'fattura_pa':
+    enable_l10n_it_edi = fields.Boolean(compute='_compute_enable_l10n_it_edi')
+    l10n_it_edi_checkbox = fields.Boolean(
+        'Tax Agency (Italy)',
+        compute='_compute_l10n_it_edi_checkbox',
+        store=True,
+        readonly=False,
+        help=(
+            "Send the invoice to the Italian Tax Agency.\n"
+            "It is set as readonly if a report has already been created, to avoid inconsistencies.\n"
+            "To re-enable it, delete the PDF attachment."
+        )
+    )
+    l10n_it_edi_readonly = fields.Boolean(compute='_compute_l10n_it_edi_readonly')
+    l10n_it_edi_warning = fields.Html(compute='_compute_l10n_it_edi_warning')
+
+    # -------------------------------------------------------------------------
+    # COMPUTE/CONSTRAINS METHODS
+    # -------------------------------------------------------------------------
+
+    @api.depends('move_ids', 'enable_l10n_it_edi')
+    def _compute_send_mail_extra_fields(self):
+        # EXTENDS account
+        super()._compute_send_mail_extra_fields()
+        for wizard in self:
+            if (wizard.enable_send_mail
+                and wizard.enable_l10n_it_edi
+                and wizard.move_ids.filtered(lambda x: x._l10n_it_edi_is_self_invoice())):
+                wizard.enable_send_mail = False
+                wizard.checkbox_send_mail = False
+                wizard.send_mail_readonly = False
+                wizard.send_mail_warning_message = None
+
+    @api.depends('move_ids', 'enable_l10n_it_edi')
+    def _compute_enable_send_by_post(self):
+        # EXTENDS snailmail_account
+        super()._compute_enable_send_by_post()
+        for wizard in self:
+            wizard.enable_send_by_post = wizard.enable_send_by_post and not (
+                wizard.enable_l10n_it_edi
+                and wizard.move_ids.filtered(lambda x: x._l10n_it_edi_is_self_invoice())
+            )
+
+    @api.depends('move_ids')
+    def _compute_enable_l10n_it_edi(self):
+        italy = self.env.ref('base.it')
+        for wizard in self:
+            enable = False
+            for move in wizard.move_ids:
+                enable |= (
+                    move.company_id.account_fiscal_country_id == italy
```

Comment:
- compare using code to avoid an unecessary extra `ref`
- create a method regrouping all the condition on journal because you will need you for the multi mode. Let's call it `_get_default_enable_l10n_it_edi`.
So, on `_compute_send_mail_extra_fields`:
`wizard.enable_l10n_it_edi = any(wizard._get_default_enable_l10n_it_edi(m) for m in wizard.move_ids)`
Note: Take care to check for PDF/your custom xml in `_get_default_enable_l10n_it_edi` for re-send flow